tests/DependencyInjection/Compiler/AddProvidersPassTest.php

```diff
@@ -84,4 +87,50 @@ public function testProcessForManyProviders()
         $providersPass = new AddProvidersPass();
         $providersPass->process($containerBuilderMock);
     }
+
+    public function testPriorityRegisteredProviders()
+    {
+        $expectedProviders = [];
+        $expectedProviders[0][] = new Reference('id');
+        $expectedProviders[20][] = new Reference('id2');
+
+        krsort($expectedProviders);
+        $expectedProviders = call_user_func_array('array_merge', $expectedProviders);
```

Comment:
you should craft the expected list directly rather than have the same code as the class that is tested.

Reply:
Hi @dbu didn't understand exactly what you mean by that?

Reply:
@emulienfou just simply that the "expected" value(s) should be as literal as possible, in this case (assuming I've had enough :coffee:):
```
$expectedProviders = [
    new Reference('id'),
    new Reference('id2'),
];
```

That way you're not testing against what we *think* might happen with, in this case, the `ksort()` and `marray_merge()`, but rather something that's easy to visually verify.

Reply:
exactly ;-)

Reply:
Oh off course thank, really needed a coffee lol